components/brave_wallet/browser/zcash/zcash_keyring.cc

```diff
@@ -17,9 +20,27 @@ ZCashKeyring::ZCashKeyring(base::span<const uint8_t> seed, bool testnet)
           seed,
           GetRootPath(testnet ? mojom::KeyringId::kZCashTestnet
                               : mojom::KeyringId::kZCashMainnet)),
-      testnet_(testnet) {}
+      testnet_(testnet) {
+#if BUILDFLAG(ENABLE_ORCHARD)
+  if (!seed.empty() && IsZCashShieldedEnabled()) {
+    auto orchard_key = HDKeyZip32::GenerateFromSeed(seed);
```

Comment:
It seems like if we pre-gen multiple orchard_keys here ahead of getting rid of the seed then we should be able to handle not needing to patch the code or get it patched upstream.

We're discussing this more in Slack already so will keep the convo there